lib/maze/schemas/trace_validator.rb

```diff
@@ -0,0 +1,85 @@
+# frozen_string_literal: true
+
+require_relative '../helper'
+
+module Maze
+  module Schemas
+
+    # Contains a set of pre-defined validations for ensuring traces are correct
+    class TraceValidator
+
+      # Whether the trace passed the validation, one of true, false, or nil (not run)
+      #   @returns [Boolean|nil] Whether the validation was successful
+      attr_reader :success
+      attr_reader :errors
+
+      # Creates the validator
+      #
+      #   @param body [Hash] The body of the trace to validate
+      def initialize(body)
+        @body = body
+        @success = nil
+        @errors = []
+      end
+
+      # Runs the validation against the trace given
+      def validate
+        @success = true
+        regex_comparison('resourceSpans.0.scopeSpans.0.spans.0.spanId', '^[A-Fa-f0-9]{16}$')
+        regex_comparison('resourceSpans.0.scopeSpans.0.spans.0.traceId', '^[A-Fa-f0-9]{32}$')
+        regex_comparison('resourceSpans.0.scopeSpans.0.spans.0.kind', '^[A-Za-z_]+')
+        regex_comparison('resourceSpans.0.scopeSpans.0.spans.0.startTimeUnixNano', '^[0-9]+$')
+        regex_comparison('resourceSpans.0.scopeSpans.0.spans.0.endTimeUnixNano', '^[0-9]+$')
+        element_contains('resourceSpans.0.resource.attributes', 'os.type')
+        element_contains('resourceSpans.0.resource.attributes', 'os.name')
+        element_contains('resourceSpans.0.resource.attributes', 'deployment.environment', 'stringValue', ['development', 'production'])
+        element_contains('resourceSpans.0.resource.attributes', 'service.name')
```

Comment:
this field is not applicable to browsers:

https://smartbear.atlassian.net/wiki/spaces/PROD/pages/3459842907/ROAD-1666+Browser+performance+-+Platforms+PD#Resource-attributes